### PR TITLE
parser, variable: fix #5478, make 'set transaction read only' works.

### LINKED ISSUE #5478: Wrong result of select @@session.tx_read_only
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```sql
mysql> select @@session.tx_read_only;
+------------------------+
| @@session.tx_read_only |
+------------------------+
|                      0 |
+------------------------+
1 row in set (0.00 sec)

mysql> set session transaction read only;
Query OK, 0 rows affected (0.00 sec)

mysql> select @@session.tx_read_only;
+------------------------+
| @@session.tx_read_only |
+------------------------+
|                      1 |
+------------------------+
1 row in set (0.00 sec)

mysql> 
```

2. What did you expect to see?
@@session.tx_read_only should be 1 after executing `set session transaction read only;`

3. What did you see instead?
@@session.tx_read_only still be 0

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
All tidb-server versions could reproduce this issue.

### PR DESCRIPTION
**Pull request by winkyao**
before this pr, use jdbc, in some case, we will get below error :

```
springboot JPA Connection is read-only. Queries leading to data modification are not allowed
```

### CODE REVIEW

**Comment by winkyao** [comment]
@coocood @shenli @tiancaiamao PTAL

File: expression/integration_test.go [hunk 1]
```diff
@@ -3101,6 +3101,25 @@ func (s *testIntegrationSuite) TestSetVariables(c *C) {
 	_, err = tk.Exec("INSERT INTO tab0 select cast('999:44:33' as time);")
 	c.Assert(err, NotNil)
 	c.Assert(err.Error(), Equals, "[types:1292]Truncated incorrect time value: '999h44m33s'")
+
+	// issue #5478
+	_, err = tk.Exec("set session transaction read write;")
+	_, err = tk.Exec("set global transaction read write;")
+	r = tk.MustQuery(`select @@session.tx_read_only, @@global.tx_read_only, @@session.transaction_read_only, @@global.transaction_read_only;`)
```
Comment (shenli) [inline expression/integration_test.go]: Is there any difference between transaction_read_only and tx_read_only?

Comment (winkyao) [reply expression/integration_test.go]: transaction_read_only was added in MySQL 5.7.20 as an alias for tx_read_only, which is now deprecated and is removed in MySQL 8.0. Applications should be adjusted to use transaction_read_only in preference to tx_read_only.

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
LGTM

**Pull request merged**